paddle/fluid/framework/executor.cc

```diff
@@ -348,8 +348,12 @@ void Executor::RunPreparedContext(ExecutorPrepareContext* ctx, Scope* scope,
       }
     }
   }
+  platform::DeviceContextPool::Instance().Get(place_)->Wait();
   if (create_vars && create_local_scope) {
     scope->DeleteScope(local_scope);
+  } else {
+    // Delete the local scopes created in operators.
+    scope->DropKids();
```

Comment:
if create_vars is False and create_local_scope is True, do we need to use
```
local_scope->DropKids()
```
?

Reply:
The `local_scope` itself is deleted by `scope->DeleteScope(local_scope);`, all the kids will be deleted, see the implementation of the destructor:

https://github.com/PaddlePaddle/Paddle/blob/0285a2b95d1c4ef30bf9a0efc19a07fc8b7a4eef/paddle/fluid/framework/scope.cc#L37-L43

Reply:
Hi @Xreki , maybe we can't delete the kids in the `execute` function, some conditions like dist training, pserver will execute the optimize blocks on the same scope instead of a new scope for each optimizes action.

related issued #10625